src/components/RatingsChart/RatingsChart.tsx

```diff
@@ -216,17 +222,39 @@ export class RatingsChart extends React.PureComponent<RatingsChartProperties, an
         return this.props.size === 0 && this.props.speed === 'overall';
     }
 
-    initialize() {{{
-        let self = this;
+    setRanges = () => {{{
         let sizes = this.chart_sizes();
-        let width = this.width = sizes.width;
+
+        this.width = sizes.width;
+        this.graph_width = 2.0 * sizes.width / 3.0;
+
+        if (this.width > 768) {  /* it gets too bunched up to show the pie */
+            this.show_pie = true;
```

Comment:
Mutable state like this should be stored in `this.state`, and ideally this should be updated as the window resizes.

This applies to any variable `width` kinda stuff too, it should be computed on render or it should be updated and stored in `this.state`.

Doing this, while extending `React.PureComponent`, lets react intelligently decide when we need to update the component.

Reply:
Does it help that the only time setRanges is called is on intitialisation and resize?

Reply:
Well, it *shouldn't* be. There's what I consider to be a bug, which is resizing the window doesn't resize the chart nor recompute the pie chart. Updating the width's and hiding/revealing the pie chart, should all be updated in an `onResize` event ideally. I didn't bring that up though because that bug existed prior to your feature, so it felt unfair to require that be fixed as part of this PR :) 

Beyond that though, just to maintain consistency if nothing else, dynamic variables should always be tracked in `this.state` as that's where other contributors know where to look for stuff. That and we get to take advantage of `PureComponent` a lot like that :)

Reply:
Eh, yeah never mind, looking at the code again it's just a funky component, most of that is my fault.